text/0096-auto-batching.md

```diff
@@ -0,0 +1,88 @@
+- Title: Auto-Batching
+
+# Auto-Batching
+
+## 1. Summary
+
+Meilisearch can automatically group consecutive asynchronous `documentAddition` or `documentPartial` tasks for the same index via an automatic batching mechanism.
+
+The user can enable this auto-batching behavior through various command flag options.
+
+## 2. Motivation
+
+We have regularly collected user pain points pointing out the slow indexing over the last year. We explained several times to users to make batches containing a maximum of documents to be updated/added to compress the indexing time of specific data structures.
+
+To make Meilisearch easier to use, we explored the idea of automatically creating these batches within Meilisearch before indexing users’ documents.
+
```

Comment:
There is another motivation actually, the feature won't only help the unexperimented users.

Currently, users can't really make batches bigger than 100mo without changing their http-conf, with the auto-batching you can actually index waaay more documents than before.